ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/HttpParser.kt

```diff
@@ -258,7 +264,7 @@ internal fun parseHeaderValue(text: CharArrayBuilder, range: MutableRange) {
         when {
             ch == HTAB || ch == ' ' -> {
             }
-            ch < ' ' -> characterIsNotAllowed(text, ch)
+            ch < ' ' -> skip = true
```

Comment:
This could potentially lead to security vulnerabilities since we simply skip all "strange" characters instead of rejecting and stopping processing the HTTP pipeline. Perhaps, we need to handle particular character codes mode precisely

Reply:
Could you explain what exact cases do you mean here? Cause we're excluding all parts with forbidden symbols from output

Reply:
What I fear is some kind of split header issue. A surprising `\n` character or something that may break HTTP pipeline processing and provide a way of spoofing

Reply:
But if we have `\n` we're proceeding to check the next header

Reply:
Ah I see, the line is already delimited so we can't face \r\n here. But what about the RFC? It clearly states that the field content is a VCHAR (0x21..0x7e) or a obs-text (0x80+). Why do we want to allow such a violation?

Reply:
Because there were multiple RFC about encoding header value and there are servers that don't encode such symbols(for cookies for example). CURL and other HTTP clients are handling that in the same way